RecoParticleFlow/PFClusterTools/src/LinkByRecHit.cc

```diff
@@ -360,6 +412,294 @@ double LinkByRecHit::testTrackAndClusterByRecHit(const reco::PFRecTrack& track,
+                        edm::soa::col::pf::track::Eta,
+                        edm::soa::col::pf::track::Phi,
+                        edm::soa::col::pf::track::Posx,
+                        edm::soa::col::pf::track::Posy,
+                        edm::soa::col::pf::track::Posz,
+                        edm::soa::col::pf::track::PosR> tracks_hcalexit_table,
+    edm::soa::TableView<edm::soa::col::pf::track::ExtrapolationValid,
+                        edm::soa::col::pf::track::Eta,
+                        edm::soa::col::pf::track::Phi,
+                        edm::soa::col::pf::track::Posx,
+                        edm::soa::col::pf::track::Posy,
+                        edm::soa::col::pf::track::Posz,
+                        edm::soa::col::pf::track::PosR> tracks_ho_table,
+    bool isBrem) {
+  //cluster position
+  const auto clustereta = cluster_table.get<pf::cluster::Eta>(icluster);
+  const auto clusterphi = cluster_table.get<pf::cluster::Phi>(icluster);
+  const auto clusterZ = cluster_table.get<pf::cluster::Posz>(icluster);
+  const auto clusterLayer = cluster_table.get<pf::cluster::Layer>(icluster);
+
+  bool barrel = false;
+  bool hcal = false;
+  // double distance = 999999.9;
+  double horesolscale = 1.0;
+
+  //track at calo's
+  double tracketa = 999999.9;
+  double trackphi = 999999.9;
+  double track_X = 999999.9;
+  double track_Y = 999999.9;
+  double track_Z = 999999.9;
+  double dHEta = 0.;
+  double dHPhi = 0.;
+
+  // Quantities at vertex
+  double trackPt = isBrem ? 999. : tracks_vtx_table.get<pf::track::Pt>(itrack);
+  // double trackEta = isBrem ? 999. : atVertex.momentum().Vect().Eta();
+
+  switch (clusterLayer) {
+    case PFLayer::ECAL_BARREL:
+      barrel = true;
+      [[fallthrough]];
+    case PFLayer::ECAL_ENDCAP:
+      // did not reach ecal, cannot be associated with a cluster.
+      if (!tracks_ecal_table.get<pf::track::ExtrapolationValid>(itrack))
+        return -1.;
+
+      tracketa = tracks_ecal_table.get<pf::track::Eta>(itrack);
+      trackphi = tracks_ecal_table.get<pf::track::Phi>(itrack);
+      track_X = tracks_ecal_table.get<pf::track::Posx>(itrack);
+      track_Y = tracks_ecal_table.get<pf::track::Posy>(itrack);
+      track_Z = tracks_ecal_table.get<pf::track::Posz>(itrack);
+
+      break;
+
+    case PFLayer::HCAL_BARREL1:
+      barrel = true;
+      [[fallthrough]];
+    case PFLayer::HCAL_ENDCAP:
+      if (isBrem) {
+        return -1.;
+      } else {
+        hcal = true;
+        // did not reach hcal, cannot be associated with a cluster.
+        if (!tracks_hcalent_table.get<pf::track::ExtrapolationValid>(itrack))
+          return -1.;
+
+        // The link is computed between 0 and ~1 interaction length in HCAL
+        dHEta = tracks_hcalexit_table.get<pf::track::Eta>(itrack) - tracks_hcalent_table.get<pf::track::Eta>(itrack);
+        dHPhi = tracks_hcalexit_table.get<pf::track::Phi>(itrack) - tracks_hcalent_table.get<pf::track::Phi>(itrack);
```

Comment:
`reco::deltaPhi`

Reply:
we kept here the old implementation with fabs explicitly to not change reco output. If one accepts a reco change already in this PR (rather than a follow-up), this can be implemented.